q2_fmt/tests/test_engraftment.py

```diff
@@ -1250,10 +1252,182 @@ def test_rename_features_with_blank_label(self):
+                                   subject_column="subject",
+                                   iterations=999)
+        real_median = np.median(stats["A:measure"].values)
+        fake_median = np.median(stats["B:measure"].values)
+        self.assertGreater(real_median, fake_median)
+
+    def test_low_donor_overlap(self):
+        metadata_df = pd.DataFrame({
+            'id': ['sample1', 'sample2', 'sample3',
+                   'donor1', 'donor2', 'donor3'],
+            'Ref': ['donor1', 'donor2', 'donor3', np.nan, np.nan,
+                    np.nan],
+            'subject': ['sub1', 'sub2', 'sub3', np.nan, np.nan,
+                        np.nan],
+            'group': [1, 1, 1, np.nan, np.nan,
+                      np.nan],
+            "Location": [np.nan, np.nan,
+                         np.nan, 'test', 'test',
+                         'test']}).set_index('id')
+
+        table_df = pd.DataFrame({
+            'id': ['sample1', 'sample2', 'sample3',
+                   'donor1', 'donor2', 'donor3'],
+            'Feature1': [1, 0, 0, 0, 1, 1],
+            'Feature2': [0, 1, 0, 1, 0, 1],
+            'Feature3': [0, 0, 1, 1, 1, 0]}).set_index('id')
+        metadata = Metadata(metadata_df)
+
+        stats, _ = peds_simulation(metadata=metadata,
+                                   table=table_df,
+                                   time_column="group",
+                                   reference_column="Ref",
+                                   subject_column="subject",
+                                   iterations=999)
+
+        real_median = np.median(stats["A:measure"].values)
+        fake_median = np.median(stats["B:measure"].values)
+        self.assertGreater(fake_median, real_median)
+
+    def test_single_donor(self):
+        metadata_df = pd.DataFrame({
+            'id': ['sample1', 'sample2', 'sample3',
+                   'donor1'],
+            'Ref': ['donor1', 'donor1', 'donor1', np.nan],
+            'subject': ['sub1', 'sub2', 'sub3', np.nan],
+            'group': [1, 1, 1, np.nan],
+            "Location": [np.nan, np.nan,
+                         np.nan, 'test']}).set_index('id')
+
+        table_df = pd.DataFrame({
+            'id': ['sample1', 'sample2', 'sample3',
+                   'donor1'],
+            'Feature1': [1, 0, 0, 0],
+            'Feature2': [0, 1, 0, 1],
+            'Feature3': [0, 0, 1, 1]}).set_index('id')
+        metadata = Metadata(metadata_df)
+
+        with self.assertRaisesRegex(AssertionError, "There is only one"
+                                    " donated microbiome in your data. *"):
+            peds_simulation(metadata=metadata,
+                            table=table_df,
+                            time_column="group",
+                            reference_column="Ref",
+                            subject_column="subject",
+                            iterations=999)
+
+    def test_mismatch_pairs(self):
```

Comment:
Rename this `test_create_mismatched_pairs`, to align the name of the test function with the name of the function being tested.